LoginApplication/app/src/main/java/com/example/webonise/loginapplication/SignUpActivity.java

```diff
@@ -0,0 +1,281 @@
+                UserProfile user = realm.createObject(UserProfile.class);
+                user.setFname(fname);
+                user.setLname(lname);
+                user.setContact(contact);
+                user.setEmail(email);
+                user.setPassword(password);
+                user.setConfirmPassword(confirmPassword);
+                user.setAddress(address);
+                user.setSecurityQuestion(securityQuestion);
+                user.setSecurityAnswer(securityAnswer);
+            }
+        });
+        Log.v("message", "Success on data storing");
+    }
+
+    private boolean valiDateProfile() {
+        String aplhaPattern = "([a-zA-Z]{3,30}\\s*)+";
+        String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
+        if (firstname.trim().isEmpty()) {
+            input_layout_fname.setError("Enter First Name");
+            return false;
+        } else {
+            input_layout_fname.setErrorEnabled(false);
+        }
+        if (!firstname.matches(aplhaPattern)) {
+            input_layout_fname.setError("Enter Valid First Name");
+            return false;
+        } else {
+            input_layout_fname.setErrorEnabled(false);
+        }
+        if (lastname.trim().isEmpty()) {
+            input_layout_lname.setError("Enter Last Name");
+            return false;
+        } else {
+            input_layout_lname.setErrorEnabled(false);
+        }
+        if (!lastname.matches(aplhaPattern)) {
+            input_layout_lname.setError("Enter Valid Last Name with atleast 3 characters");
+            return false;
+        } else {
+            input_layout_lname.setErrorEnabled(false);
+        }
+        if (contactString.length() > 10 || contactString.length() < 10) {
+            input_layout_contact.setError("Enter 10 Digit Contact Number");
+            return false;
+        } else {
+            input_layout_contact.setErrorEnabled(false);
+        }
+        if (email.trim().isEmpty()) {
+            input_layout_email.setError("Enter Email ID");
+            return false;
+        } else {
+            input_layout_email.setErrorEnabled(false);
+        }
+        if (!email.matches(emailPattern)) {
+            input_layout_email.setError("Enter Valid Email ID");
+            return false;
+        } else {
+            input_layout_email.setErrorEnabled(false);
+        }
+        if (password.trim().isEmpty()) {
+            input_layout_password.setError("Enter Password");
+            return false;
+        } else {
+            input_layout_email.setErrorEnabled(false);
+        }
+        if (confirmPassword.trim().isEmpty()) {
+            input_layout_confirmPassword.setError("Enter Confirm Password");
+            return false;
+        } else {
+            input_layout_email.setErrorEnabled(false);
+        }
+
+        if (!password.equals(confirmPassword)) {
+            input_layout_confirmPassword.setError("Enter Password same as in Password Field");
```

Comment:
Shouldn't use hardcoded string values. Take them out in strings.xml and use reference